Shared/Samples/List contents of KML file/ListContentsOfKMLFileView.swift

```diff
@@ -0,0 +1,261 @@
+        /// A dataset containing the KML data from a local file.
+        @Published private(set) var kmlDataset: KMLDataset?
+        
+        /// The viewpoints for the nodes in the dataset.
+        @Published private(set) var nodeViewpoints: [String: Viewpoint] = [:]
+        
+        /// The error shown in the error alert.
+        @Published var error: Error?
+        
+        /// A scene for displaying the KML data.
+        let scene: ArcGIS.Scene = {
+            let scene = Scene(basemapStyle: .arcGISImagery)
+            let elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
+            scene.baseSurface.addElevationSource(elevationSource)
+            return scene
+        }()
+        
+        /// The task used for the asynchronous setup operations.
+        private var setupTask: Task<Void, Error>?
+        
+        init() {
+            setupTask = Task { [weak self] in
+                guard let self else { return }
+                
+                do {
+                    try await setUpKMLDataset()
+                } catch {
+                    self.error = error
+                }
+            }
+        }
+        
+        deinit {
+            setupTask?.cancel()
+        }
+        
+        /// Sets up the KML dataset and adds it to the scene as layer.
+        private func setUpKMLDataset() async throws {
+            // Creates the dataset using a local ".kml" file in the bundle.
+            let kmlDataset = KMLDataset(name: "esri_test_data", bundle: .main)!
+            try await kmlDataset.load()
+            self.kmlDataset = kmlDataset
+            
+            // Adds the dataset to the scene as a KML layer.
+            let kmlLayer = KMLLayer(dataset: kmlDataset)
+            scene.addOperationalLayer(kmlLayer)
+            try await scene.load()
+            
+            try await setUpKMLNodes(kmlDataset.rootNodes)
+        }
+        
+        /// Recursively creates viewpoints for KML nodes in a given list.
+        /// - Parameter kmlNodes: The list of KML nodes to set up.
+        private func setUpKMLNodes(_ kmlNodes: [KMLNode]) async throws {
+            // Loads the surface so that the elevation can be queried when the viewpoint is made.
+            try await scene.baseSurface.load()
+            
+            for node in kmlNodes {
+                let viewpoint = try await Viewpoint(kmlNode: node, surface: scene.baseSurface)
+                nodeViewpoints[node.name] = viewpoint
+                
+                // Ensures the node is visible since some are hidden by default.
+                node.isVisible = true
+                
+                if let childNodes = node.children {
+                    try await setUpKMLNodes(childNodes)
+                }
+            }
+        }
+    }
+}
+
+// MARK: Helper Extensions
+
+extension KMLNode {
+    /// The child nodes of the node, if any.
+    fileprivate var children: [KMLNode]? {
```

Comment:
I used `fileprivate` instead of `private` extensions like we usually do since it was suggested by Xcode 16's new swift-format feature. It suggested other stuff too, but I wasn't sure about those since they seemed to conflict with "Re-Indent".

Reply:
On closer inspection, I found that these caused linter warnings, so I removed them to just go with the status quo for now.